gateway-sp-comms/src/single_sp/update.rs

```diff
@@ -257,57 +258,137 @@ fn read_auxi_check_from_tlvc(data: &[u8]) -> Result<[u8; 32], UpdateError> {
+                        // At the time of this writing `image-name` is a recent addition to
+                        // hubris archives, so skip this check if we don't have one.
+                        Err(HubtoolsError::MissingFile(..)) => (),
+                        Err(err) => return Err(err.into()),
+                    }
+
+                    // TODO: Add a caboose BORD preflight check just like the SP has, once the
+                    // RoT has a caboose and we have RPC calls to read its values.
+                    rot_image
+                }
+                _ => return Err(UpdateError::InvalidSlotIdForOperation),
+            }
         }
-    }
-
-    // Preflight check 1: Does the image name of this archive match the target
-    // slot?
-    match archive.image_name() {
-        Ok(image_name) => match (image_name.as_str(), slot) {
-            ("a", 0) | ("b", 1) => (), // OK!
-            _ => return Err(UpdateError::RotSlotMismatch { slot, image_name }),
-        },
-        // At the time of this writing `image-name` is a recent addition to
-        // hubris archives, so skip this check if we don't have one.
-        Err(HubtoolsError::MissingFile(..)) => (),
-        Err(err) => return Err(err.into()),
-    }
+        SpComponent::STAGE0 => {
+            // Staging area for a Bootloader image:
+            // stage0next can be updated directly, stage0 cannot.
+            // The RoT will reject updates to slot !=1 but don't
+            // waste its time.
+            if slot != 1 {
+                return Err(UpdateError::InvalidSlotIdForOperation);
+            }
 
-    // TODO: Add a caboose BORD preflight check just like the SP has, once the
-    // RoT has a caboose and we have RPC calls to read its values.
+            RawHubrisArchive::from_vec(image.clone())
+                .and_then(|archive| archive.image.to_binary())
+                .or_else(|hubtool_error|
+                    // Prior to v1.2.0, Bootleby was packaged as a simple
+                    // zip archive containing a "bootleby.bin" file.
+                    //
+                    // TODO: Remove support for the old image format when
+                    // those bootleby versions are no longer used in
+                    // manufacturing and rollback protection can be used to
+                    // prevent their re-introduction. Until then, we need to
+                    // be able to test update and rollback using the oldest
+                    // releases that may be in customers' racks or spares pool.
+                    bootleby_from_old_style_archive(image, log)
+                        // Report the original Hubtools error if
+                        // this second chance did not work.
+                        .map_err(|_| hubtool_error))?
+
+            // TODO: Even though the RoT will protect itself, put
+            // pre-flash checks here for BORD, Bootloader vs Hubris,
+            // and signature validity.
```

Comment:
Do you want to do this TODO as part of this PR or a followup?

Reply:
That should be a follow-up. [Issue 221](https://github.com/oxidecomputer/management-gateway-service/issues/221)